### DISCUSSION TITLE
PyRFC - Decimal conversion error

### DISCUSSION THREAD

**Discussion by cauashmith** [SAP/PyRFC#378] [Q&A]
Hello guys, y'all doing good?

I need some help..
I'm are trying to call a SAP RFC but i'm are having "class decimal.ConversionSyntax" as return.
I've also noticed that this RFC calling only works when the libs are from SAPGUI, but, my goal is to make it work with Windows dependencies (NWRFCSDK).

The following images shows the dependencies for Windows OS that i've installed, and the setted enviroment variable in my PC.

![nwrfcsdk](https://github.com/user-attachments/assets/3e4235bb-09f4-40af-8dfb-3280e89c5652)
![enviroment_var](https://github.com/user-attachments/assets/8e7746c5-90b6-4c70-86ee-d3a447048d31)

Any help is useful.
Regards, Cauã.

**Comment by danielhillding** [comment]
Hi Cauã, 

I have the same issue. Have you been able to solve it yet? 

Regards, 
Daniel

**Reply by hkroeber** [reply]
Hi, 
please have a look at #371. The newsest PyRFC lib requries your client has set the locale "en_GB.UTF-8" when working with numeric strings.
Hagen

**Reply by danielhillding** [reply]
Thank you, however it still gives me the same error even though i set the locale to "en_GB.UTF-8". My code is a simple test program: 

    `conn = Connection(**params)
    print("Connected to SAP `successfully.")
    response = conn.call(
        'BAPI_MATERIAL_GET_DETAIL',
        MATERIAL= "TEST",
        )
    print(response)`

**Reply by cauashmith** [reply]
Hi Daniel,

I found an Issue opened by Sebastian after a long search, saying that they'll not maintain this project anymore due to changing priorities. As you might've see, the latest versions of this project is built with an older RFC SDK version, which is no longer supported by SAP.

So, to make it work, you have to use an older version from SAP LIB (like the ones from SAP GUI). 

Issue that explains it better: (https://github.com/SAP/PyRFC/issues/372) 

Regards, Cauã.

**Reply by NdBuk** [reply]
> 
> So, to make it work, you have to use an older version from SAP LIB (like the ones from SAP GUI).
> 

Yeah, but according to SAP note **3302936 - Versions of SAP NetWeaver RFC SDK Library available for download**, only the LATEST version is available for download.... struggling to find a copy of patch 12 for Red Hat anywhere...

**Reply by cauashmith** [reply]
Yeah...

My initial idea was to make my code works on a Windows OS then migrate to a Red Hat by changing the RFC SDK libs, but isn't possible because only the latest version is available for download, as you just said, and they're having this maintance Issues.

But actually, you can find some non-official SDK's that could possibly make it work (it's hard to find, i've already tried), but that's not the path that i would follow.

If you find any solution, answer it, please.

Regards, Cauã.

**Reply by StephDam** [reply]
Just removed the Decimal conversion for BCD fields (I return them as strings), recompiled the module, and I do not have the error anymore. You can inject data in Pandas and convert to numeric the corresponding columns.

**Comment by davidemenghini** [comment]
Hello everyone,
Do you have found solution? In my codebase I suppressed the decimal.ConverionSyntax error when I call a specific custom rfc. Do you have any other more elegant solutions? I would like to avoid to remove the decimal.ConverionSyntax error.
